src/Microsoft.AspNetCore.Http/Internal/DefaultHttpRequest.cs

```diff
@@ -151,12 +156,19 @@ public override Task<IFormCollection> ReadFormAsync(CancellationToken cancellati
             return FormFeature.ReadFormAsync(cancellationToken);
         }
 
+        public override RouteValueDictionary RouteValues
```

Comment:
We should avoid returning a null value here. If the feature doesn't exist, or the value is null then an empty collection should be created, stored, and returned. We do something similar for cookies, query, etc..